microsoft-edge/extensions-chromium/developer-guide/minimize-page-load-time-impact.md

```diff
@@ -0,0 +1,251 @@
+
+If your extension is small enough, you won't need a build tool to split your code. If your extension is larger, and if your code is more complex to manage, use a build to split your code into smaller chunks. Build tools can help you organize your code into logical units, which can be loaded on demand. For example, you can use webpack to split your code into _entry points_ and _dynamic imports_:
+
+* Entry points are loaded on every page load.
+* Dynamic imports are only loaded on demand, for example when the user interacts with the webpage or with the extension's UI:
+
+  ```javascript
+  // When the user clicks on the page.
+  document.addEventListener("click", async () => {
+    // Dynamically load the code that's needed to handle the click event.
+    const module = await import("chunk.js");
+    // Do something with the newly loaded module code.
+  });
+  ```
+
+
+<!-- ====================================================================== -->
+## Only load your content script in required pages and frames
+
+Your extension might not need to run on every webpage that the user visits. To reduce the amount of code that runs when webpages load, configure your extension to load content scripts only on the pages and frames where they are needed.
+
+To configure the pages and frames where your content scripts are loaded, define URL patterns in your extension manifest file by using the `matches` property in the `content_scripts` section. To learn more, see [Inject scripts](https://developer.chrome.com/docs/extensions/develop/concepts/content-scripts#functionality) in the Chrome Extensions documentation.
+
+You can also use the `chrome.scripting` extensions API to programmatically inject your content script into the webpage. This API allows you to inject your content script based on the user's actions, the webpage content, or the extension logic. To learn more, see [chrome.scripting](https://developer.chrome.com/docs/extensions/reference/api/scripting) in the Chrome Extensions documentation.
+
+Use the following best practices when configuring where your content scripts are loaded:
+
+* Use the most specific URL patterns possible for the `matches` and `exclude_matches` properties in your extension manifest file. For example, if your content script only needs to run on webpages of the example.com domain, use `https://example.com/*` instead of "`*://*/*`.
+* To control whether your content script only runs in the top-level frame, or also in nested frames of the webpage that matches a URL pattern, use the `all_frames` property in your extension manifest file. By default, this property is set to false, which means that your content script will only run in the top-level frame. If your content script needs to access or modify the DOM in nested frames, set this property to true. Note that setting `all_frames` to true increases the amount of code that runs on a webpage.
```

Comment:
```suggestion
* To control whether your content script only runs in the top-level frame, or also in nested frames of the webpage that matches a URL pattern, use the `all_frames` property in your extension manifest file. By default, this property is set to `false`, which means that your content script will only run in the top-level frame. If your content script needs to access or modify the DOM in nested frames, set this property to true. Note that setting `all_frames` to true increases the amount of code that runs on a webpage.
```
check if `false` is technically correct key word, if so, backticks adds clarity.  might be instead 0, eg
@captainbrosset

Reply:
added backticks